Subject: [PATCH] readdir: return value missed in getdents64
X-BK-ChangeSetKey: pmeda@akamai.com[torvalds]|ChangeSet|20050108214136|41821
Date:	Sat, 08 Jan 2005 23:14:54 +0000
Patch-mainline: yes
From: olh@suse.de


ChangeSet 1.2273, 2005/01/08 13:41:36-08:00, pmeda@akamai.com

	[PATCH] readdir: return value missed in getdents64
	
	Small issue: return value missed in getdents64, but handled in getdents.
	
	Signed-Off-by: Prasanna Meda <pmeda@akamai.com>
	Signed-off-by: Andrew Morton <akpm@osdl.org>
	Signed-off-by: Linus Torvalds <torvalds@osdl.org>



 readdir.c |    3 ++-
 1 files changed, 2 insertions(+), 1 deletion(-)


diff -Nru a/fs/readdir.c b/fs/readdir.c
--- a/fs/readdir.c	2005-01-08 15:09:44 -08:00
+++ b/fs/readdir.c	2005-01-08 15:09:44 -08:00
@@ -287,8 +287,9 @@
 	lastdirent = buf.previous;
 	if (lastdirent) {
 		typeof(lastdirent->d_off) d_off = file->f_pos;
-		__put_user(d_off, &lastdirent->d_off);
 		error = count - buf.count;
+		if (__put_user(d_off, &lastdirent->d_off))
+			error = -EFAULT;
 	}
 
 out_putf:
-
To unsubscribe from this list: send the line "unsubscribe bk-commits-head" in
the body of a message to majordomo@vger.kernel.org
More majordomo info at  http://vger.kernel.org/majordomo-info.html

